core/src/main/java/com/cryptape/cita/protocol/system/CITAjSystemContract.java

```diff
@@ -593,9 +663,13 @@ public QueryResourceResult queryResource(String fromAddr, String permissionAddr)
         List<TypeReference<?>> outputParamters
                 = Collections.singletonList(new TypeReference<DynamicArray<Address>>() {});
         List<Type> resultTypes = CITASystemContract.decodeCallResult(callResult, outputParamters);
+        if (resultTypes.isEmpty())
+            return null;
+        List<String> list = new ArrayList<>();
         ArrayList<Address> results = (ArrayList<Address>) resultTypes.get(0).getValue();
-        List<String> list = new ArrayList<>(results.size());
         for (Address address : results) {
```

Comment:
can be replaced with Java8 lambda expression: 
```
List<String> list = results.stream()
  .filter(address -> 
    (!INTRA_GROUP_USER_MANAGEMENT_ADDR.equalsIgnoreCase(address.getValue()) )
  .map(Address::getValue).collect(Collectors.toList())
```

Reply:
@yangzhenlong this is not a feature PR, this is a "sync" PR, no need to review the code quality in this case, FYI.